### PR TITLE
[REF-1464] Handle requirements.txt encoding

### PR DESCRIPTION
**Pull request by martinxu9**
### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/reflex-dev/reflex/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/reflex-dev/reflex/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### New Feature Submission:

- [x] Does your submission pass the tests? 
- [x] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### **After** these steps, you're ready to open a pull request.
1. Previously the handling of `requirements.txt` in user local environment assumes `utf-8` the default encoding used in builtin open method. If user has a different encoding, we open and append will cause the file to have inconsistent encoding on different lines and make part of the content unreadable using any given encoding
2. Add UT to cover the non utf-8 files.
3. For hosting: now that we re-check/re-generate requirements.txt when user deploys, it is safe to simply pin the reflex version instead of >=.

### CODE REVIEW

File: pyproject.toml
Comment (masenf) [inline pyproject.toml:61]: just out of curiosity, was `chardet` considered for this? it's what i've mostly seen used in the past, but i see it hasn't had a release for quite some time.

**Review by masenf — DISMISSED** [review]

File: pyproject.toml
Comment (martinxu9) [reply pyproject.toml:61]: Not really, I didn't look into that. I saw some online answers using this, and this seems to be a dependency of some of current packages already.
But I will double check what people say about those two before merging.

Comment (masenf) [reply pyproject.toml:61]: oh nice, charset-normalizer _already_ being in our installed payload makes this an easy decision

Comment (martinxu9) [reply pyproject.toml:61]: Oh ok. Let's use this package for now. It's in a rather tiny area of our code

File: pyproject.toml
Comment (martinxu9) [inline pyproject.toml:86]: got a pyright error about connection not having a commit() function, but this makes it go away

Comment (martinxu9) [reply pyproject.toml]: removing it seems to not cause a problem. I guess local system pre-commit probably doesn't sync up with poetry?

**Comment by martinxu9** [comment]
@masenf thumbs up one more time please

**Review by masenf — APPROVED** [review]
sorry, thought this already merged

**Comment by martinxu9** [comment]
> sorry, thought this already merged

No worries. I think our branch is set up to dismiss approvals if there's any new changes. I needed to merge upstream

**Pull request merged**
